### ISSUE TITLE
Tests are failing in a clean clone

### ISSUE THREAD

**Issue by srijan** [greyorange/mnesia_migrate#5]
```
$ make deps
...
$ make eunit SKIP_DEPS=1
 GEN    test-dir
 GEN    eunit
db_migration_tests:8: check_file_creation_test_ (Check Src Dir Exists)...*failed*
in function db_migration_tests:'-check_file_creation_test_/0-fun-0-'/1 (test/db_migration_tests.erl, line 8)
**error:{assertEqual,[{module,db_migration_tests},
              {line,8},
              {expression,"filelib : is_dir ( db_migration : get_migration_source_filepath ( ) )"},
              {expected,true},
              {value,false}]}
  output:<<"">>

db_migration_tests:10: check_file_creation_test_ (Check Beam Dir Exists)...*failed*
in function db_migration_tests:'-check_file_creation_test_/0-fun-2-'/1 (test/db_migration_tests.erl, line 10)
**error:{assertEqual,[{module,db_migration_tests},
              {line,10},
              {expression,"filelib : is_dir ( db_migration : get_migration_beam_filepath ( ) )"},
              {expected,true},
              {value,false}]}
  output:<<"">>

db_migration_tests: check_migration_file_creation_test_ (check revision file creation)...*failed*
in function db_migration_tests:'-check_migration_file_creation_test_/0-fun-0-'/2 (test/db_migration_tests.erl, line 33)
in call from db_migration_tests:'-check_migration_file_creation_test_/0-fun-2-'/0 (test/db_migration_tests.erl, line 33)
**error:{assertEqual,[{module,db_migration_tests},
              {line,33},
              {expression,"filelib : is_file ( Filename )"},
              {expected,true},
              {value,false}]}
  output:<<"Compile file: /home/srijan/workspace/mnesia_migrate/schema.template
schema.template: Warning: Compiled template not saved (need out_dir option)
Base Rev module is ''
New file created "a2436b29c_migration.erl"
">>

db_migration_tests: check_migration_file_creation_test_ (check revision file creation and deletion)...*failed*
in function db_migration_tests:'-check_migration_file_creation_test_/0-fun-3-'/1 (test/db_migration_tests.erl, line 43)
**error:{assertCmd_failed,[{module,db_migration_tests},
                   {line,43},
                   {command,"rm /home/gaurav/project/mnesia_migrate/src/migrations/ad5900fcb_migration.erl"},
                   {expected_status,0},
                   {status,1}]}
  output:<<"Compile file: /home/srijan/workspace/mnesia_migrate/schema.template
Base Rev module is ''
New file created "ad5900fcb_migration.erl"
">>

db_migration_tests: check_database_related_function_test_ (check_applied_head_test_1)...*failed*
in function db_migration:get_applied_head/0 (src/db_migration.erl, line 139)
in call from db_migration_tests:'-check_database_related_function_test_/0-fun-0-'/1 (test/db_migration_tests.erl, line 54)
**error:{badmatch,{aborted,{no_exists,schema_migrations}}}
  output:<<"">>

db_migration_tests: check_db_related_test_ (check_applied_head_test_2)...*failed*
in function db_migration:get_applied_head/0 (src/db_migration.erl, line 139)
in call from db_migration_tests:'-check_db_related_test_/0-fun-0-'/1 (test/db_migration_tests.erl, line 65)
**error:{badmatch,{aborted,{no_exists,schema_migrations}}}
  output:<<"">>

=======================================================
  Failed: 6.  Skipped: 0.  Passed: 5.
erlang.mk:6244: recipe for target 'eunit' failed
make: *** [eunit] Error 2
```

**Comment by srijan** [comment]
- [x] All default directories should be relative
- [x] All default directories should exist in a fresh clone (so that tests can pass)
- [x] Schema migrations should be created inside `get_applied_head/0` if they don't already exist

**Comment by keeper575-zz** [comment]
Fixed by e0f4077 and 873fff5

**Comment by keeper575-zz** [comment]
@srijan could you update here when all test cases are working properly in clean clone ?

**Comment by srijan** [comment]
Two tests still failing.

```
$ make distclean
...
$ make deps
...
$ make eunit SKIP_DEPS=1
 GEN    test-dir
 GEN    eunit
db_migration_tests: check_database_related_function_test_ (check_applied_head_test_1)...*failed*
in function db_migration:init_migrations/0 (src/db_migration.erl, line 49)
in call from db_migration_tests:'-check_database_related_function_test_/0-fun-1-'/0 (test/db_migration_tests.erl, line 53)
**throw:{error,{bad_type,schema_migrations,disc_copies,nonode@nohost}}
  output:<<"Table schema_migration not found, creating...
mnesia create table error: {bad_type,schema_migrations,disc_copies,
                                     nonode@nohost}
">>

db_migration_tests: check_db_related_test_ (check_applied_head_test_2)...*failed*
in function db_migration:init_migrations/0 (src/db_migration.erl, line 49)
in call from db_migration_tests:'-check_db_related_test_/0-fun-1-'/0 (test/db_migration_tests.erl, line 65)
**throw:{error,{bad_type,schema_migrations,disc_copies,nonode@nohost}}
  output:<<"Table schema_migration not found, creating...
mnesia create table error: {bad_type,schema_migrations,disc_copies,
                                     nonode@nohost}
">>

=======================================================
  Failed: 2.  Skipped: 0.  Passed: 9.
erlang.mk:6244: recipe for target 'eunit' failed
make: *** [eunit] Error 2
```

**Comment by srijan** [comment]
I added travis ci file so that every push will be build.

Now you can directly check if CI is failing or passing.

**Comment by keeper575-zz** [comment]
Fixed in e30bffc3b6b87758f
